### PR TITLE
Feat(BCE-29): Export Report Button

### PR DESCRIPTION
**Pull request by Funnyadd**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/BCE-29)
## Context:
Added export button to export chart reports in a pdf format with a graph, table and all the necessary information related to the report and its content.

## Changes
- Added dynamic export button for each row on the chart report table
- Added pdf template to create the pdf needed
- Added create, fetch and deletion functions for the chart reports pdf file
- Added tests for:
  1. chartReportDao
  2. reportService
  3. reportController

#### BEFORE
image

#### AFTER
image

#### Test Proof
image

note: The tests that are not 'done' will have to stay like this for the github actions to work.
         Also, teh pdfTemplate can't really be tested as it depends on the data received on many levels

## Dev notes (Optional)
- We might want to change the chart bar colors on the dashboard to match with the newly selected colors used in the chart report pdf template.

### CODE REVIEW

File: client/src/pages/Reports.js [hunk 1]
```diff
@@ -47,6 +50,73 @@ const Reports = () => {
...
+                            }
+                        }
+                        else if (error.request) {
+                            alert("Could not fetch pdf file...");
+                        }
+                    });
+                }
+                else {
+                    alert("Could not reach b&C Engine...");
+                }
```
Comment (cjayneb) [inline client/src/pages/Reports.js:102]: if `response` comes back as a falsy value(the last else here), it does not mean that the B&C Engine was not reached. However, if the code execution reaches the `else if(error.request)`, that means that the B&C Engine could not be reached.

File: client/src/pages/Reports.js [hunk 2]
```diff
@@ -47,6 +50,73 @@ const Reports = () => {
...
+                    if (error.response.status === 403 || error.response.status === 401) {
+                        alert("You are not authorized to perform this action.");
+                    }
+                    else {
+                        alert("Could not generate pdf file...");
+                    }
+                }
+                else if (error.request) {
+                    alert("Could not generate pdf file...");
+                }
```
Comment (cjayneb) [inline client/src/pages/Reports.js:116]: Same comment as above

File: server/tests/controllers/report.controller.test.js [hunk 3]
```diff
@@ -27,6 +28,7 @@ let reportServiceSpy = jest.spyOn(ReportService, 'getChartReportsByUserId')
     }));
 
 const makeApp = require('../../app');
+const { UUID } = require('sequelize');
```
Comment (cjayneb) [inline server/tests/controllers/report.controller.test.js:31]: unused import

**Review by cjayneb — APPROVED** [review]
Looks good to me. Good job making that Chart Report template. it looks like it was painful. Just check my comments and merge in brother!

File: server/tests/controllers/report.controller.test.js (hunk 3 above)
Comment (Funnyadd) [reply server/tests/controllers/report.controller.test.js]: removed with current commit

File: client/src/pages/Reports.js (hunk 1 above)
Comment (Funnyadd) [reply client/src/pages/Reports.js:102]: What would you recommend I put instead?

File: client/src/pages/Reports.js (hunk 2 above)
Comment (Funnyadd) [reply client/src/pages/Reports.js:116]: What would you recommend I put instead?

File: client/src/pages/Reports.js (hunk 1 above)
Comment (cjayneb) [reply client/src/pages/Reports.js:102]: in the last else which is highlighted here, I would just put 'Could not fetch Chart Report' or something like that

File: client/src/pages/Reports.js (hunk 2 above)
Comment (cjayneb) [reply client/src/pages/Reports.js:116]: here when there is an error with the request (`error.request`), we should let the user know by makiing an alert that says 'Could not reach the b&C Engine'

**Review by NickEspitalier — APPROVED** [review]
Perfect, and great implementation for the pdf.

LGTM!

**Pull request merged**